csharp/src/Microsoft.ML.OnnxRuntime/Training/TrainingSession.shared.cs

```diff
@@ -351,6 +475,14 @@ private void Init(SessionOptions sessOptions, CheckpointState state, byte[] trai
                     _trainOutputNames.Add(GetOutputName(i, true));
                 }
 
+                _trainInputNames = new List<string>();
+                UIntPtr inputCount = UIntPtr.Zero;
+                NativeApiStatus.VerifySuccess(NativeTrainingMethods.OrtGetTrainingModelOutputCount(_nativeHandle, out inputCount));
```

Comment:
```suggestion
                NativeApiStatus.VerifySuccess(NativeTrainingMethods.OrtGetTrainingModelInputCount(_nativeHandle, out inputCount));
```

Reply:
Thank you for catching this. I would have hated if this got missed and shipped. :)